docs/man/man3/ParseStrings.3
.TH "ParseStrings" 3 "Thu May 25 2017" "Version 0.2" "SensorNode" \" -*- nroff -*-
.ad l
.nh
.SH NAME
ParseStrings \- \fBParseStrings\fP class definition\&.  

.SH SYNOPSIS
.br
.PP
.PP
\fC#include <ParseStrings\&.h>\fP
.SS "Public Member Functions"

.in +1c
.ti -1c
.RI "void \fBParse\fP (const char *string)"
.br
.RI "\fIParse const character string\&. \fP"
.ti -1c
.RI "void \fBParseMAC\fP ()"
.br
.RI "\fIParse MAC address\&. \fP"
.ti -1c
.RI "uint64_t \fBgetMAC\fP ()"
.br
.RI "\fIGet MAC address as uint64_t\&. \fP"
.ti -1c
.RI "void \fBgetPropertyString\fP (char *propString)"
.br
.RI "\fIGet Property string\&. \fP"
.ti -1c
.RI "void \fBgetPropertyIDAsMAC\fP (uint8_t propString[8])"
.br
.RI "\fIget Property ID as a MAC address array \fP"
.ti -1c
.RI "uint8_t \fBgetCmdID\fP ()"
.br
.RI "\fIget Command ID \fP"
.ti -1c
.RI "uint8_t \fBgetCmdProperty\fP ()"
.br
.RI "\fIReturns command property\&. \fP"
.ti -1c
.RI "void \fBgetPropertyID\fP (char *PIDString)"
.br
.RI "\fIReturns Property ID as character array\&. \fP"
.ti -1c
.RI "void \fBPrintVars\fP (char *string)"
.br
.RI "\fIReturns Variables as string\&. \fP"
.ti -1c
.RI "void \fBPrintMAC\fP (char *string)"
.br
.RI "\fIReturns MAC address to string\&. \fP"
.ti -1c
.RI "void \fBClearAll\fP ()"
.br
.RI "\fIClears all variables\&. \fP"
.ti -1c
.RI "\fBParseStrings\fP ()"
.br
.RI "\fI\fBParseStrings\fP standard constructor\&. \fP"
.ti -1c
.RI "\fBParseStrings\fP (char *string)"
.br
.RI "\fI\fBParseStrings\fP constructor\&. \fP"
.in -1c
.SS "Private Attributes"

.in +1c
.ti -1c
.RI "uint8_t \fBcmdID\fP"
.br
.RI "\fICommand ID\&. \fP"
.ti -1c
.RI "uint8_t \fBcmdP\fP"
.br
.RI "\fICommand Property\&. \fP"
.ti -1c
.RI "char \fBPID\fP [36]"
.br
.RI "\fIProperty ID\&. \fP"
.ti -1c
.RI "char \fBpString\fP [36]"
.br
.RI "\fIProperty String\&. \fP"
.ti -1c
.RI "uint8_t \fBaddr\fP [5]"
.br
.RI "\fI5 Byte address array \fP"
.in -1c
.SH "Detailed Description"
.PP 
\fBParseStrings\fP class definition\&. 

Defines methods for accessing the \fBParseStrings\fP interface 
.SH "Constructor & Destructor Documentation"
.PP 
.SS "ParseStrings::ParseStrings ()"

.PP
\fBParseStrings\fP standard constructor\&. new \fBParseStrings\fP object and initializes variables 
.SS "ParseStrings::ParseStrings (char * string)"

.PP
\fBParseStrings\fP constructor\&. 
.PP
\fBParameters:\fP
.RS 4
\fIstring\fP Character array to parse 
.RE
.PP

.SH "Member Function Documentation"
.PP 
.SS "void ParseStrings::ClearAll ()"

.PP
Clears all variables\&. 
.SS "uint8_t ParseStrings::getCmdID ()"

.PP
get Command ID 
.PP
\fBReturns:\fP
.RS 4
Command ID as uint8_t 
.RE
.PP

.SS "uint8_t ParseStrings::getCmdProperty ()"

.PP
Returns command property\&. 
.PP
\fBReturns:\fP
.RS 4
Command Property as uint8_t 
.RE
.PP

.SS "uint64_t ParseStrings::getMAC ()"

.PP
Get MAC address as uint64_t\&. 
.PP
\fBReturns:\fP
.RS 4
MAC address as a 64 Bit unsigned integer 
.RE
.PP

.SS "void ParseStrings::getPropertyID (char * PIDString)"

.PP
Returns Property ID as character array\&. 
.PP
\fBParameters:\fP
.RS 4
\fIPIDString\fP Character array of Property ID 
.RE
.PP

.SS "void ParseStrings::getPropertyIDAsMAC (uint8_t propString[8])"

.PP
get Property ID as a MAC address array 
.PP
\fBParameters:\fP
.RS 4
\fIpropString\fP 8 Byte Address array 
.RE
.PP

.SS "void ParseStrings::getPropertyString (char * propString)"

.PP
Get Property string\&. 
.PP
\fBParameters:\fP
.RS 4
\fIpropString\fP Character array 
.RE
.PP

.SS "void ParseStrings::Parse (const char * string)"

.PP
Parse const character string\&. 
.PP
\fBParameters:\fP
.RS 4
\fIstring\fP Character string to parse 
.RE
.PP

.SS "void ParseStrings::ParseMAC ()"

.PP
Parse MAC address\&. Parses MAC address from pString into the addr array 
.SS "void ParseStrings::PrintMAC (char * string)"

.PP
Returns MAC address to string\&. 
.PP
\fBParameters:\fP
.RS 4
\fIstring\fP Character array of MAC address 
.RE
.PP

.SS "void ParseStrings::PrintVars (char * string)"

.PP
Returns Variables as string\&. 
.PP
\fBParameters:\fP
.RS 4
\fIstring\fP Character array of variable values 
.RE
.PP

.SH "Member Data Documentation"
.PP 
.SS "uint8_t ParseStrings::addr[5]\fC [private]\fP"

.PP
5 Byte address array 
.SS "uint8_t ParseStrings::cmdID\fC [private]\fP"

.PP
Command ID\&. 
.SS "uint8_t ParseStrings::cmdP\fC [private]\fP"

.PP
Command Property\&. 
.SS "char ParseStrings::PID[36]\fC [private]\fP"

.PP
Property ID\&. 
.SS "char ParseStrings::pString[36]\fC [private]\fP"

.PP
Property String\&. 

.SH "Author"
.PP 
Generated automatically by Doxygen for SensorNode from the source code\&.


src/IO/ParseStrings.cpp
/**
****************************************************************************
* Project: SensorNode
*
* ParseStrings.cpp
*
* Created on: 15 Dec 2016
*
* Author: Matthias Minx
*
* Revision: 0.2
*
*
****************************************************************************/
/** \file ParseStrings.cpp
 *
 */


#include <avr/io.h>
#include <stdint.h>
#include <avr/pgmspace.h>

#include "ParseStrings.h"
#include <stdlib.h>
#include <string.h>
#include <inttypes.h>
#include <stdio.h>
//#include "USART.h"



ParseStrings::ParseStrings() {
	// TODO Auto-generated constructor stub
	//u = USART();
	ClearAll();


}
ParseStrings::ParseStrings(char* string) {
	// TODO Auto-generated constructor stub

	//u = USART();
	Parse(string);



}

void ParseStrings::ClearAll(){

	cmdID = 0;
	cmdP = 0;
	sprintf(PID,"%s"," ");
	sprintf(pString,"%s"," ");



}

void ParseStrings::ParseMAC(){


	char *addToken;
	//strtoken(pString,(char*)":",addToken);

	uint8_t i = 0;
	char *tmp;
	char tmpStr[36];
	sprintf(tmpStr,pString);
	//strdup(pString);


//	u.writeString(pString);
	//u.writeString("\r\n");
	addToken = strtok_r(tmpStr, ":",&tmp);

//	u.writeString(addToken);

	addr[0] = (uint8_t)strtoul(addToken,NULL,16);

	  while ( addToken ) {
	    i++;
	    addToken = strtok_r(NULL, ":",&tmp);
	    if(addToken!=NULL && i<5){
	    	addr[i] = (uint8_t)strtoul(addToken,NULL,16);
	    }


	  }
	//  u.writeString(pString);
	//  free(tmpStr);

}

void ParseStrings::getPropertyString(char *propString){
	sprintf(propString,"%s",pString);


}
void ParseStrings::getPropertyIDAsMAC(uint8_t propString[8]){

	char *addToken;
		//strtoken(pString,(char*)":",addToken);

	uint8_t i = 0;
	char *tmp;
	char tmpStr[36];
	sprintf(tmpStr,PID);



	addToken = strtok_r(tmpStr, ":",&tmp);

	propString[0] = (uint8_t)strtoul(addToken,NULL,16);

	while ( addToken ) {
		i++;
		addToken = strtok_r(NULL, ":",&tmp);
		if(addToken!=NULL && i<8){
			propString[i] = (uint8_t)strtoul(addToken,NULL,16);
		}


	}

	//free(tmpStr);




}


uint64_t ParseStrings::getMAC(){

uint64_t value = reinterpret_cast<uint64_t>(&addr[0]) |
		reinterpret_cast<uint64_t>(&addr[1]) << 8 |
		reinterpret_cast<uint64_t>(&addr[2]) << 16 |
		reinterpret_cast<uint64_t>(&addr[3]) << 24 |
		reinterpret_cast<uint64_t>(&addr[4]) << 32;




	return (value);

}

uint8_t ParseStrings::getCmdID(){
	return (cmdID);

}
uint8_t ParseStrings::getCmdProperty(){
	return (cmdP);
}
void ParseStrings::getPropertyID(char *PIDString){
	sprintf(PIDString,"%s",PID);
}


void ParseStrings::Parse(const char* string){

	ClearAll();

	char *token;
	char* p;
	uint8_t i = 0;
	char *tmp;
	char *tmpStr = strdup(string);

//strtoken(s,delimiter,token);

//	u.writeString((char*)string);
//	u.writeString(" | ");
//	u.writeString(tmpStr);
//	u.writeString(" |t:  ");

	  token = strtok_r((char*)string, "|",&tmp);

	// u.writeString(token);
//	 u.writeString(" | ");
	  cmdID = uint8_t(strtoul(token,&p,16)/(0x10))*0x10;

	  cmdP = uint8_t(strtoul(token,&p,16))-cmdID;



	//  char t[50];
	//  sprintf(t," ");
	//  sprintf(t," cmdID: %i | cmdP: %i \r\n",cmdID,cmdP);
	//  u.writeString(t);


	  while ( token ) {
		i++;
	    token = strtok_r(NULL, "|",&tmp);
//	    u.writeString(token);
//	 u.writeString(" . ");
	    if(token!=NULL){
	    	if(i==1){
	    		sprintf(PID,"%s",token);
	    	//	u.writeString("\t");
	    	//	u.writeString(PID);
	    	//	PID = atoi(token);
	    	}else if(i==2){
	    		sprintf(pString,"%s",token);
	    	}


	    }

	  }

	  free(tmpStr);



}


void ParseStrings::PrintVars(char* string){

	char buff[100];
	sprintf(string,"\r\n##### Config String ###### \r\n");

//	sprintf(buff,"MAC: %02X:%02X:%02X:%02X:%02X \r\n",addr[0],addr[1],addr[2],addr[3],addr[4]);strcat(string,buff);
	sprintf(buff,"pString: %s \r\n",pString); strcat(string,buff);

	sprintf(buff,"cmdID: %02X \r\n",getCmdID()); strcat(string,buff);
	sprintf(buff,"cmdP: %02X \r\n",getCmdProperty());strcat(string,buff);
	sprintf(buff,"PID: %s \r\n",PID);strcat(string,buff);
	sprintf(buff,"######################### \r\n");strcat(string,buff);



}

void ParseStrings::PrintMAC(char* string){

	char buff[100];

	sprintf(buff,"Pipe: %s MAC: %02X:%02X:%02X:%02X:%02X \r\n",PID,addr[0],addr[1],addr[2],addr[3],addr[4]);strcat(string,buff);

	sprintf(buff,"######################### \r\n");strcat(string,buff);



}
